search-app/src/cmr/search/services/acls/granule_acls.clj

```diff
@@ -152,22 +152,25 @@
         (gc/or-conds [concept-ids-cond entry-titles-cond])
         (or concept-ids-cond entry-titles-cond)))))
 
-
 (defn collection-identifier->query-condition
   "Converts an acl collection identifier to an query condition. Switches implementations based
   on whether the user's query contained collection ids. This implementation assumes the query-coll-ids
   passed in are limited to the provider of the collection identifier."
   [context query-coll-ids provider-id collection-identifier]
   (let [colls-in-prov-cond (provider->collection-condition query-coll-ids provider-id)]
-    (if-let [{:keys [entry-titles access-value temporal]} collection-identifier]
+    (if-let [{:keys [concept-ids entry-titles access-value temporal]} collection-identifier]
       (let [entry-titles-cond (provider+entry-titles->collection-condition
                                 context query-coll-ids provider-id entry-titles)
+            concept-ids-cond (provider->collection-condition concept-ids provider-id)
             access-value-cond (some-> (access-value->query-condition access-value)
                                       q/->CollectionQueryCondition)
             temporal-cond (some-> temporal temporal->query-condition q/->CollectionQueryCondition)]
-
         ;; If there's no entry title condition the other conditions must be scoped by provider
-        (gc/and-conds (remove nil? [(or entry-titles-cond colls-in-prov-cond)
+        ;; In anycase we want to use concept ids instead of entry-titles, because entry-titles can and
+        ;; do frequently change, so to prevent temporary loss of permissions concept ids are used
+        ;; when present.  There really shouldn't be a case where entry-tiles exist and concept-ids do
+        ;; not.
+        (gc/and-conds (remove nil? [(or concept-ids-cond entry-titles-cond)
```

Comment:
The feels wrong. We are dropping the `colls-in-prov-cond` which is constructed from the query params from the search. Why?